tests/key_field_sample_generated.h

```diff
@@ -0,0 +1,262 @@
+// automatically generated by the FlatBuffers compiler, do not modify
+
+
+#ifndef FLATBUFFERS_GENERATED_KEYFIELDSAMPLE_KEYFIELD_SAMPLE_H_
+#define FLATBUFFERS_GENERATED_KEYFIELDSAMPLE_KEYFIELD_SAMPLE_H_
+
+#include "flatbuffers/flatbuffers.h"
+
+// Ensure the included flatbuffers.h is the same version as when this file was
+// generated, otherwise it may not be compatible.
+static_assert(FLATBUFFERS_VERSION_MAJOR == 22 &&
+              FLATBUFFERS_VERSION_MINOR == 10 &&
+              FLATBUFFERS_VERSION_REVISION == 26,
+             "Non-compatible flatbuffers version included");
+
+namespace keyfield {
+namespace sample {
+
+struct Baz;
+
+struct Bar;
+
+struct FooTable;
+struct FooTableBuilder;
+
+FLATBUFFERS_MANUALLY_ALIGNED_STRUCT(1) Baz FLATBUFFERS_FINAL_CLASS {
+ private:
+  uint8_t a_[4];
+  uint8_t b_;
+
+ public:
+  Baz()
+      : a_(),
+        b_(0) {
+  }
+  Baz(uint8_t _b)
+      : a_(),
+        b_(flatbuffers::EndianScalar(_b)) {
+  }
+  Baz(flatbuffers::span<const uint8_t, 4> _a, uint8_t _b)
+      : b_(flatbuffers::EndianScalar(_b)) {
+    flatbuffers::CastToArray(a_).CopyFromSpan(_a);
+  }
+  const flatbuffers::Array<uint8_t, 4> *a() const {
+    return &flatbuffers::CastToArray(a_);
+  }
+  bool KeyCompareLessThan(const Baz *o) const {
+    for (auto i = 0; i < a()->size(); i++){ 
+      if (a()->Get(i) != o->a()->Get(i)) return a()->Get(i) < o->a()->Get(i);
+    }
+    return false;
+  }
+  int KeyCompareWithValue(const uint8_t _a[4]) const { 
+    for (auto i = 0; i < a()->size(); i++) {
+      if(a()->Get(i) != _a[i]) 
```

Comment:
I would save a()->Get(i) and _a[i] in a local variable, since you use it three times in this function.